### PR TITLE
✨  Allow contents: write for Token-Permissions when doing mvn release 

### PR DESCRIPTION
**Pull request by katzj**
#### What kind of change does this PR introduce?

This allows workflow steps that call 'mvn release:prepare' to use `contents: write` permissions as they are needed for doing the maven release process.

(Is it a bug fix, feature, docs update, something else?)

Somewhere between a bug and a feature

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

Current behavior is that a workflow that is doing mvn release will get dinged on the Token-Permissions check due to `contents: write` permission for that workflow

#### What is the new behavior (if this is a feature change)?**

A workflow using `mvn release:prepare` will be allowed the `contents: write` permission without lowering the score

- [X] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

NONE

#### Does this PR introduce a user-facing change?

```release-note
The Token-Permissions check no longer penalizes workflows with the `contents: write` permission if they are using it to run `mvn release:prepare`.
```

### CODE REVIEW

File: checks/raw/permissions.go [hunk 1]
```diff
@@ -491,6 +491,16 @@ func isReleasingWorkflow(workflow *actionlint.Workflow, fp string, pdata *permis
 			},
 			LogText: "candidate SLSA publishing workflow using slsa-github-generator",
 		},
+		{
+			// Running mvn release:prepare requires committing changes.
+			// https://maven.apache.org/maven-release/maven-release-plugin/examples/prepare-release.html
+			Steps: []*fileparser.JobMatcherStep{
+				{
+					Run: ".*mvn.*release:prepare.*",
```
Comment (laurentsimon) [inline checks/raw/permissions.go:499]: should this be `mvnw` instead of `mvn`?

**Review by laurentsimon — APPROVED** [review]
Thank you, this is great!

File: checks/raw/permissions.go (hunk 1 above)
Comment (katzj) [reply checks/raw/permissions.go:499]: Not everyone uses mvnw; some people just depend on an installed mvn instead. The regex will allow both to work

Comment (laurentsimon) [reply checks/raw/permissions.go:499]: Got it, thanks.

**Pull request merged**